enhancements/console/samples.md

```diff
@@ -0,0 +1,396 @@
+
+All previously supported samples opens a simplified "Import from Git" flow and prefill the Git repository.
+
+When a `GitImport` sample is selected, the same form will be opened and prefilled.
+
+1. The Git `url` is required.
+2. A Git `ref` is optional and can be used to clone a specific branch, tag or commit.
+
+There is no other attribute supported at the moment. It supports only public Git repositories.
+
+```yaml
+  source:
+    type: GitImport
+    gitImport:
+      url: https://github.com/jboss-openshift/openshift-quickstarts
+      ref: master # optional
+```
+
+#### `ContainerImport`
+
+When the user selects a `ContainerImport` sample, the **"Add"** > **"Container images"** page will be opened
+and the given container image is prefilled as "Image name from external registry".
+
+Other options like the Resource type (Deployment, DeploymentConfig or Serverless Deployment) will
+be selected based on the cluster defaults and the user preferences.
+
+```yaml
+  source:
+    type: ContainerImport
+    containerImport:
+      image: registry.access.redhat.com/ubi8/ubi-minimal:8.8-860
+```
+
+### Localization
+
+The `ConsoleSample` resources provide the same [optional localization labels as `ConsoleQuickStarts`](https://github.com/openshift/enhancements/blob/master/enhancements/console/quick-starts.md#quick-start-internationalization).
+
+The `console.openshift.io/name: explore-serverless` label specifies the name of the sample. This should be consistent across all translations of a given sample,
+as it allows us to avoid showing the same Quick Start content in multiple languages.
+
+If a sample is not translated into a user's language, the English version will be shown.
+
+See the [internationalization](https://github.com/openshift/enhancements/blob/master/enhancements/console/internationalization.md) and [QuickStart enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/console/quick-starts.md) for more information.
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+  labels:
+    console.openshift.io/name: java-sample # optional, metadata.name is also fine
+spec:
+  title: Java sample
+  abstract: Build and run Java applications using Maven and OpenJDK.
+  description: |
+    # About this sample
+    Build and run Java applications using Maven and OpenJDK.
+    # Links
+    * [Sample repository](https://github.com/jboss-openshift/openshift-quickstarts)
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: Source to Image
+  tags:
+  - java
+  - jboss
+  - openjdk
+  source:
+    type: GitImport
+    gitImport:
+      url: https://github.com/jboss-openshift/openshift-quickstarts
+---
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-samples-de
+  labels:
+    console.openshift.io/lang: de
+    console.openshift.io/name: java-sample # must match the console.openshift.io/name label or metadata.name above
```

Comment:
This is an existing mechanism that is already implemented right? Not something new?

Would probably have been good to include this information in the spec rather than as labels

Reply:
In fact, it's not implemented for Quick Starts, but it's part of the Quick Start enhancement doc.

@spadgett wdyt?

Reply:
We discussed this a while when working on quick starts, but I forget the entire context. I believe labels were preferred since we can query on them and exclude the languages we don't need from the results. This could be significant since we're base64-encoding icons in these resources, and they might get large. I think using `spec` would be cleaner from an API standpoint, though.

@christianvogt Do you happen to remember?

Reply:
The label filtering is a good argument to be fair, but from an API standpoint, yes, I think this would be better as an API field